### PR TITLE
Quarkus 1.6.0

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>

### What does this PR do?

- Introduces Quarkus extension 1.6.0

Devfile to test

```

---
apiVersion: 1.0.0
metadata:
  name: java-quarkus-1.6.0
projects:
  -
    name: java-web-spring
    source:
      type: git
      location: "https://github.com/spring-projects/spring-petclinic.git"
      commitId: 4953f87917b449a404a7f1f4e2457836b9eafbbc

components:
  -
    type: chePlugin
    reference: >-
      https://raw.githubusercontent.com/vitaliy-guliy/che-plugin-registry/quarkus-1.6.0/v3/plugins/redhat/quarkus-java11/1.6.0/meta.yaml
    preferences:
      java.project.importOnFirstTimeStartup: 'automatic'

```

### CODE REVIEW

**Comment by svor** [comment]
@ericwill should we update [vscode-extensions.json](https://github.com/eclipse/che-plugin-registry/blob/master/vscode-extensions.json#L243) by hand?

**Comment by ericwill** [comment]
> @ericwill should we update [vscode-extensions.json](https://github.com/eclipse/che-plugin-registry/blob/master/vscode-extensions.json#L243) by hand?

Yes.

File: v3/plugins/redhat/quarkus-java11/1.6.0/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+displayName: Quarkus Tools with Java 11
+title: Quarkus Tools by Red Hat
+description: This plug-in provides support for Quarkus development via a Quarkus language server, Java jdt.ls and Java Debugger.
+icon: https://raw.githubusercontent.com/redhat-developer/vscode-quarkus/v1.6.0/icons/logo.png
+repository: https://github.com/redhat-developer/vscode-quarkus
+category: Language
+firstPublicationDate: "2020-08-18"
+spec:
+  containers:
+    - image: "quay.io/eclipse/che-sidecar-java:11-f76ca45"
```
Comment (ericwill) [inline v3/plugins/redhat/quarkus-java11/1.6.0/meta.yaml:15]: Please use the latest che-sidecar-java:11 image listed in the [vscode-extension.json file](https://github.com/eclipse/che-plugin-registry/blob/master/vscode-extensions.json) for the quarkus extension.

**Comment by benoitf** [comment]
I'm a bit worried that by bumping vscode-java, users will have a default prompt about importing their java projects

is there a way to apply the configuration from meta.yaml ? and not to patch every devfile by adding

```
    preferences:
      java.project.importOnFirstTimeStartup: 'automatic'
```

**Comment by vitaliy-guliy** [comment]
> I'm a bit worried that by bumping vscode-java, users will have a default prompt about importing their java projects
> 
> is there a way to apply the configuration from meta.yaml ? and not to patch every devfile by adding
> 
> ```
>     preferences:
>       java.project.importOnFirstTimeStartup: 'automatic'
> ```

I tried to do it before, but wasn't lucky.
Adding preferences from plugin is not supported and it doesn't described in the plugin spec
https://github.com/eclipse/che-plugin-registry#plugin-meta-yaml-structure

But.. we can write a plugin. Which will read all the plugin meta.yamls and add the corresponding preferences if such present.
Or just extend the functionality of workspace plugin.

**Comment by vitaliy-guliy** [comment]
@ericwill done with vscode-extensions.json and with container version

**Comment by vitaliy-guliy** [comment]
[ci-test]

**Comment by ericwill** [comment]
Do we need the latest version of `vscode-java`? IMO it shouldn't be newer than what we already have in Che, unless absolutely necessary.

**Pull request merged**